Framework/PythonInterface/plugins/functions/ChudleyElliot.py

```diff
@@ -32,12 +35,22 @@ def function1D(self, xvals):
         tau = self.getParameterValue("Tau")
         length = self.getParameterValue("L")
         xvals = np.array(xvals)
+
         with np.errstate(divide='ignore'):
             hwhm = self.hbar*(1.0 - np.sin(xvals * length)
                               / (xvals * length))/tau
-
         return hwhm
 
+    def functionDeriv1D(self, xvals, jacobian):
+        tau = self.getParameterValue("Tau")
+        length = self.getParameterValue("L")
+
+        for i, x in enumerate(xvals, start=0):
+            s = 1.0 - math.sin(x*length)/(x*length)
```

Comment:
`math.sin(x*length)/(x*length)` produces a singularity at `x = 0` which shouldn't be there. The correct function to use here would be `numpy.sinc()` (see [Wikipedia](https://en.wikipedia.org/wiki/Sinc_function) and [numpy docs](https://docs.scipy.org/doc/numpy-1.7.0/reference/generated/numpy.sinc.html#numpy.sinc)): `numpy.sinc(x*length/numpy.pi)`.